### PR TITLE
Transformers 4.44 support

### PR DESCRIPTION
**Pull request by IlyasMoutawwakil**
# What does this PR do?

Following https://github.com/huggingface/optimum/pull/1996

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by IlyasMoutawwakil** [comment]
two failures remain from the custom remote code in `glm4` and `chatglm` model repos @eaidova

**Comment by eaidova** [comment]
@IlyasMoutawwakil  I updated models code https://huggingface.co/katuni4ka/tiny-random-chatglm2/commit/de19195f343ff61c96a25b973c8f800542802bb0
https://huggingface.co/katuni4ka/tiny-random-glm4/commit/44354258a5292224c98979c97c5622f2538e53b6

please rerun

File: optimum/intel/ipex/modeling_base.py [hunk 1]
```diff
@@ -523,7 +523,7 @@ def _prepare_past_key_values(self, input_ids):
                 ]
             )
             return past_key_values
-        elif model_type == "bloom":
+        elif model_type == "bloom" and is_transformers_version("<", "4.44"):
```
Comment (eaidova) [inline optimum/intel/ipex/modeling_base.py:526]: just from curiosity, what changed in bloom in transformers 4.44?

Comment (IlyasMoutawwakil) [reply optimum/intel/ipex/modeling_base.py:526]: attention block modeling was changed for the sake of compatibility with the cache classes https://github.com/huggingface/transformers/pull/31445#pullrequestreview-2122697114

Comment (eaidova) [reply optimum/intel/ipex/modeling_base.py:526]: @IlyasMoutawwakil what about the second part of my comment? What if user has old model converted with previous transformers version and new transformers installed, such case will not work now.

Comment (IlyasMoutawwakil) [reply optimum/intel/ipex/modeling_base.py:526]: is there a way to check the version used for exporting the model ?
if there is we can use the custom modeling or or raise an error asking to re export.

Comment (eaidova) [reply optimum/intel/ipex/modeling_base.py:526]: yes, it is possible to check using rt_info:

```
from optimum.intel.openvino import OVModelForCausalLM
ov_model = OVModelForCausalLM.from_pretrained("bigscience/bloom-560m", export=True)
transformers_version = ov_model.model.rt_info["optimum"]["transformers_version"].value
```

File: optimum/intel/openvino/modeling_decoder.py [hunk 2]
```diff
@@ -404,7 +405,8 @@ def prepare_inputs(
         **kwargs,
     ) -> Dict:
         batch_size = input_ids.shape[0]
-        if self.config.model_type == "bloom":
+        model_transformers_version = Version(self.model.rt_info["optimum"]["transformers_version"].value)
```
Comment (eaidova) [inline optimum/intel/openvino/modeling_decoder.py:408]:
```suggestion
        model_transformers_version = Version(self.model.rt_info["optimum"]["transformers_version"].value if "optimum" in self.model.rt_info[ else "0.0.0")
```

**Review by eaidova — APPROVED** [review]

**Review by echarlaix — APPROVED** [review]
LGTM, thanks @IlyasMoutawwakil !

File: setup.py [hunk 3]
```diff
@@ -28,8 +28,8 @@
 
 INSTALL_REQUIRE = [
     "torch>=1.11",
-    "transformers>=4.36.0,<4.44.0",
-    "optimum@git+https://github.com/huggingface/optimum.git",
+    "transformers>=4.36,<4.45",
+    "optimum@git+https://github.com/huggingface/optimum.git@transformers-4.44",
```
Comment (echarlaix) [inline setup.py:32]:
```suggestion
    "optimum@git+https://github.com/huggingface/optimum.git",
```

**Pull request merged**